### PR TITLE
Add docker image build to travis CI script.

### PR DESCRIPTION
**Pull request by douglas-reid**
This change is [Reviewable](https://reviewable.io/reviews/istio/mixer/194)

### CODE REVIEW

**Comment by douglas-reid** [comment]
NOT for review at this time.  This PR exists as a placeholder for testing perms issues, etc., that will impact the ability to push images to a location.

**Comment by douglas-reid** [comment]
PTAL. I believe that I've got the auth bits setup correctly now. With this setup, the docker image should only be updated on merges. For now, it uses a service account on the istio-test project to push to GCR.

This should address the final bit of #162.

File: docker/travis_gcloud.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+set -ev
+if [ "${TRAVIS_PULL_REQUEST}" = "false" ]; then
+	openssl aes-256-cbc -K $encrypted_2f660428f0db_key -iv $encrypted_2f660428f0db_iv -in istio-test.json.enc -out istio-test.json -d
```
Comment (mandarjog) [inline docker/travis_gcloud.sh]: are the $encrypted_2f660428f0db_key set in env vars for the build?

Comment (douglas-reid) [inline docker/travis_gcloud.sh]: yes. as best I can tell, they were added when I used the travis CLI to encrypt the key file.

**Pull request merged**